cunumeric/array.py

```diff
@@ -3147,14 +3178,55 @@ def mean(
         else:
             return sum_array
 
+    @add_boilerplate()
+    def _nanmean(
+        self,
+        axis: Any = None,
+        dtype: Union[np.dtype[Any], None] = None,
+        out: Union[ndarray, None] = None,
+        keepdims: bool = False,
+        where: Union[ndarray, None] = None,
+    ) -> ndarray:
+        from . import _ufunc
+
+        if np.issubdtype(dtype, np.integer) or np.issubdtype(dtype, np.bool_):
+            return self.mean(
+                axis=axis, dtype=dtype, out=out, keepdims=keepdims, where=where
+            )
+
+        if dtype is None:
+            dtype = self.dtype
+
+        nan_mask = _ufunc.bit_twiddling.bitwise_not(
+            _ufunc.floating.isnan(self)
+        )
+        normalizer = nan_mask.sum(axis=axis, keepdims=keepdims, where=where)
+        sum_array = self._nansum(axis, keepdims=keepdims, where=where)
+        sum_array = sum_array.astype(dtype)
+        if dtype.kind == "f" or dtype.kind == "c":
+            sum_array.__itruediv__(
+                np.array(normalizer, dtype=dtype),
+            )
+        else:
+            sum_array.__ifloordiv__(np.array(normalizer, dtype=dtype))
+
+        if out is not None and sum_array is not out:
+            if out.dtype != sum_array.dtype:
+                out._thunk.convert(sum_array._thunk)
+            else:
+                out = sum_array
+            return out
+        else:
+            return sum_array
```

Comment:
This implementation is missing some special cases that `mean` is taking care of, e.g. `float64` intermediate and return values should be used for integer inputs. Can we just fall back to `mean`?

```suggestion
        nan_mask = _ufunc.bit_twiddling.bitwise_not(
            _ufunc.floating.isnan(self)
        )
        if where is not None:
            nan_mask &= where
        return self.mean(
            axis=axis, dtype=dtype, out=out, keepdims=keepdims, where=nan_mask
        )
```